datumaro/plugins/coco_format/extractor.py

```diff
@@ -54,145 +59,134 @@ def __init__(self, path, task, *,
 
         self._merge_instance_polygons = merge_instance_polygons
 
+        json_data = self._load_json(path)
+        self._label_map = {} # coco_id -> dm_id
```

Comment:
This is mostly for comment. I agree with your points, but as a developer, it's not something I would like to worry about. Also, it will produce wrong field access error messages, if there are annotations, but no categories.